src/Maestro/tests/Maestro.ScenarioTests/AsyncDisposable.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Threading.Tasks;
+
+namespace Maestro.ScenarioTests
+{
+    public class AsyncDisposable : IAsyncDisposable
+    {
+        private readonly Func<ValueTask> _dispose;
+
+        public static IAsyncDisposable Create(Func<ValueTask> dispose)
+        {
+            return new AsyncDisposable(dispose);
+        }
+
+        private AsyncDisposable(Func<ValueTask> dispose)
+        {
+            _dispose = dispose;
+        }
+
+        public ValueTask DisposeAsync()
+        {
+            return _dispose();
```

Comment:
Should we make it safe against mutli-dispose?  Or are we going to assume the caller will make their delegate properly multi-dispose safe?

Reply:
I'll do it.